sympy/core/basic.py

```diff
@@ -1623,32 +1623,70 @@ def count_ops(self, visual=None):
         from sympy import count_ops
         return count_ops(self, visual)
 
-    def doit(self, **hints):
-        """Evaluate objects that are not evaluated by default like limits,
-        integrals, sums and products. All objects of this kind will be
-        evaluated recursively, unless some species were excluded via 'hints'
-        or unless the 'deep' hint was set to 'False'.
-
-        >>> from sympy import Integral
+    def doit(self, deep=True, properties=[], **hints):
+        """Evaluate objects that are not evaluated like operations, limits,
+        integrals, sums and products.
+
+        Arguments will be evaluated recursively, unless some species were
+        excluded via 'properties' or unless the 'deep' hint was set to 'False'.
```

Comment:
"species" is not a common term used in SymPy (do you mean classes?). Also "properties" is not a very clear parameter name.

Reply:
Well, I believe the original author ment "classes" indeed. I'll change it.
I borrowed the name "properties" from `Wild`, but have been uneasy with this name as well. Any suggestions?